### ISSUE TITLE
sign on loss function

### ISSUE THREAD

**Issue by johnwlambert** [IshmaelBelghazi/ALI#14]
Hi, in the paper pseudocode, for Loss_d and Loss_g, the gradient ascent is turned into gradient descent by making the whole loss negative (negative signs in front of every term).

However, [in the code](https://github.com/IshmaelBelghazi/ALI/blob/master/ali/bricks.py#L69), I don't see any of those negative signs. What am I missing?

Thank you!

**Comment by IshmaelBelghazi** [comment]
The negative term in front of the discriminator loss is factored in the softplus.

**Comment by edgarriba** [comment]
@johnwlambert check the expanded formulas

![formulas](https://cloud.githubusercontent.com/assets/5157099/25168792/1d9bab94-24e5-11e7-8a10-99fa1254ab68.jpg)

**Comment by edgarriba** [comment]
whereas `softplus(x) = log(1 + exp(x))`
